Sources/Ads/AdTracker.swift

```diff
@@ -0,0 +1,150 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  AdTracker.swift
+//
+//  Created by RevenueCat on 1/15/25.
+
+import Foundation
+
+#if ENABLE_AD_EVENTS_TRACKING
+
+/**
+ Tracks ad-related events to RevenueCat.
+
+ Use this class to report ad impressions, clicks, and revenue to RevenueCat alongside your subscription data.
+ This enables comprehensive LTV tracking across subscriptions and ad monetization.
+
+ ## Usage
+
+ Access the ad tracker through the `Purchases` singleton:
+
+ ```swift
+ let adTracker = Purchases.shared.adTracker
+ ```
+
+ ## Example
+
+ ```swift
+ // Track an ad impression
+ await Purchases.shared.adTracker.trackAdDisplayed(.init(
+     networkName: "AdMob",
+     mediatorName: .appLovin,
+     placement: "home_screen",
+     adUnitId: "ca-app-pub-123",
+     adInstanceId: "instance-456"
+ ))
+
+ // Track ad revenue
+ await Purchases.shared.adTracker.trackAdRevenue(.init(
+     networkName: "AdMob",
+     mediatorName: .appLovin,
+     placement: "home_screen",
+     adUnitId: "ca-app-pub-123",
+     adInstanceId: "instance-456",
+     revenueMicros: 1500000,  // $1.50
+     currency: "USD",
+     precision: .exact
+ ))
+ ```
+ */
+@available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+@objc(RCAdTracker)
```

Comment:
I see we have the `@objc` marker here, but it's not in the `track` methods or in the `AdEventData` types: `AdDisplayed`, `AdOpened` or `AdRevenue`. These are structs and cannot be ported to Objective-C as is.

So, do we want to have these APIs be compatible with Objective-C? 
* If not, we can (and should) remove the `@objc` marker
* If yes, then we might need to rethink these objects to make them `NSObject` subclasses instead of `struct`s (it pains me to propose this 😿)

Reply:
Per our convo the other day, I'll tackle this in a separate PR dedicated to this goal. Marking resolved.